### PR TITLE
Expose index points in IndexedGzipFile via points method

### PR DESCRIPTION
**Pull request by szborows**
Hi,
thanks for the library - it's brilliant!

My use case involves retrieving part of a gzip file from the server and reconstructing it locally with sparse files. In order to know which region of the archive has to be downloaded I need to store entry points in a database. Then I can do HTTP with Range header.

This is minimal change that does what I want (that I could come up with :smile: ). Feel free to challenge it- I know you might have different opinion on this.
If you think this change is not justified I can stick to my fork in the dependencies.

### CODE REVIEW

**Comment by szborows** [comment]
Obviously I will document it once we agree on the overall interface.

**Comment by pauldmccarthy** [comment]
Hi @szborows, thanks for the PR! I don't quite understand your use case - would you mind elaborating a little more (or linking to source, if it's public)?

 - You have a web app which is managing access to a large `gz` file
 - The web app generates a full index into the file, and stores the seek point locations in a database
 - You receive HTTP requests for specific parts of the file

What I don't understand is - couldn't you do this with just the knowledge of the uncompressed file size? Once an `IndexedGzipFile` has built its index, you don't need to provide specific seek point locations - you can seek to any offset. I may be misunderstanding things though.

**Comment by szborows** [comment]
Hi @pauldmccarthy , sure I can elaborate. It's not public, but the scenario is as follows.
1. Big GZIP archives are stored on S3.
2. There's a backend service that retrieves part of these files with HTTP Range header. It also retrieves GZIP header.
3. The service creates sparse file with ftruncate and pastes both header and archive region.
4. Then, it decompresses as if it was real archive, but it's a little bit faked. Finally file is sent to the user.

So if I have offsets and sizes for uncompressed data I have no idea which region of the archive to download from S3.

Hopefully you can understand it now. Pic attached. If there is easier way to achieve that then please let me know.

ftrunc12

**Comment by pauldmccarthy** [comment]
Aah, I think I understand - you want to read a specific range of the compressed data, so you need to know where the seek points are - cool!

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.coverage.*
+*.so
+*.c
+__pycache__
+*.gz*
```
Comment (pauldmccarthy) [inline .gitignore:5]: Mind removing this file from the PR?

**Comment by szborows** [comment]
That's right.

File: indexed_gzip/tests/ctest_indexed_gzip.pyx [hunk 2]
```diff
@@ -609,6 +609,23 @@ def test_iter(drop):
                  next(f)
 
 
+def test_index_points():
```
Comment (pauldmccarthy) [inline indexed_gzip/tests/ctest_indexed_gzip.pyx:612]: Thanks for the unit test .. would you be able to extend it to a scenario where multiple seek points are generated?

File: indexed_gzip/indexed_gzip.pyx [hunk 3]
```diff
@@ -289,6 +289,10 @@ cdef class _IndexedGzipFile:
 
         return proxy()
 
+    def points(self):
```
Comment (pauldmccarthy) [inline indexed_gzip/indexed_gzip.pyx:292]: How about naming the method something a little more descriptive, e.g. `def seek_points`?

**Comment by szborows** [comment]
@pauldmccarthy I don't see your UT remark in Files changed tab so I can't reply there.
Can you look now whether this is something you're looking for?

**Comment by pauldmccarthy** [comment]
Perfect - thanks! I'm happy to add in a doc-string. I'll release a new version with this change over the next couple of days (unless the CI jobs have broken again...).

**Pull request merged**